src/main/groovy/org/transmartproject/batch/highdim/i2b2/DummyFactGenerator.groovy

```diff
@@ -62,7 +65,12 @@ class DummyFactGenerator extends AbstractItemCountingItemStreamItemReader<Clinic
 
     @Override
     protected void doOpen() throws Exception {
-        maxItemCount = mappingsFileRowStore.rows.size()
+        subjectConceptMappingFileRows = mappingsFileRowStore
+            .rows
+            .groupBy { [it.subjectId, it.conceptFragment] }
+            .values()
```

Comment:
`groupBy` doesn't guarantee an ordering, but the class requires a consistent order of the elements after a job restart. Throw in a `sort`.